### PR TITLE
fix for issue #69 - allowing values for features

### PR DESCRIPTION
**Pull request by ekandrotA9**
*Issue #, if available:*
#69 

*Description of changes:*
Removed the lines of code labeled Hack, that check for values greater than 10 then assign it a zero.
Verified that the parsing to .nc files writes out a data section when the analog flag is set.
Verified that MovieLen still works correctly (which doesn't use analog, to verify that this didn't impact non-analog case).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ekandrotA9** [comment]
The comments for the BatchNorm:

This has been implemented using cuDNN for the actual BN code.  Currently, it has only been hooked up to fully connected layers, because there was an immediate use case for it.  I will create a bug for the task of adding BN to the other two types of supported layers (there are printfs in the code right now where to insert these changes).  I also changed the cifar-10 example to have the fully connected layer use BN.  It had a dramatic impact on the learning rate and the recall.

The main issue with cuDNN is that it seems to be non-deterministic.  This made it difficult to verify tweaks to learning rates, because of the variance between reruns on the same inputs.  I have tuned the numbers to have a large change, but once cuDNN is fixed or replaced it can be fine tuned even more.

For this change, I'm looking for a review to how well this fits into DSSTNE.  I believe I handled memory correctly - I had to move my allocations from Allocate to NNLayer::NNLayer because the layer description used during learning is only around during layer creation, and also Allocate calls Deallocate, which would blow away any values read into the BN memory structures.

The BN part of this is trivial, since it is handled by a library.  A poorly documented library, so if someone knows how it actually works, they could verify that I figured it out correctly.  I believe I did, and results indicate that all is well.

There are a lot of functions in DSSTNE and a lot of code flow - I had to keep some allocations and setting of variables local to cuDNN calls in order to be sure things were set when needed.

Let me know if there are any bugs that I can fix with the code.  Thank you for your help.
